test/e2e/common/pods.go

```diff
@@ -879,25 +879,27 @@ var _ = framework.KubeDescribe("Pods", func() {
 		}
 		ginkgo.By("creating a Pod with a static label")
 		_, err := f.ClientSet.CoreV1().Pods(testNs).Create(context.TODO(), &testPod, metav1.CreateOptions{})
-		framework.ExpectNoError(err, "failed to create Pod")
+		framework.ExpectNoError(err, "failed to create Pod " + testPodName + " in namespace " + testNs)
 
 		ginkgo.By("setting up a watch for the Pod")
 		podWatchTimeoutSeconds := int64(180)
 		podWatch, err := f.ClientSet.CoreV1().Pods(testNs).Watch(context.TODO(), metav1.ListOptions{LabelSelector: testPodLabelsFlat, TimeoutSeconds: &podWatchTimeoutSeconds})
-		framework.ExpectNoError(err, "failed to set up watch")
+		framework.ExpectNoError(err, "failed to set up watch for Pod " + testPodName + " in namespace " + testNs)
 
 		podWatchChan := podWatch.ResultChan()
 		ginkgo.By("watching for Pod to be ready")
 		eventFound := false
 		for watchEvent := range podWatchChan {
 			podWatchEvent, ok := watchEvent.Object.(*v1.Pod)
-			framework.ExpectEqual(ok, true, "unable to assert runtime object type to v1.Pod")
+			if ok == false {
+				continue
+			}
 			if podWatchEvent.Status.Phase == v1.PodRunning {
```

Comment:
Consider
```
if e, ok := watchEvent.Object.(*v1.Pod); ok {
  if podWatchEvent.Status.Phase == v1.PodRunning {
    ...
  }
}
```
or
```
podWatchEvent, ok := watchEvent.Object.(*v1.Pod)
if ok && podWatchEvent.Status.Phase == v1.PodRunning {
...
}
```

Reply:
The last appears the best to me; Updated in a3b3d8902c3